libtiledbsoma/src/soma/soma_geometry_column.cc

```diff
@@ -0,0 +1,365 @@
+#include "soma_geometry_column.h"
+
+namespace tiledbsoma {
+
+std::shared_ptr<SOMAGeometryColumn> SOMAGeometryColumn::create(
+    std::shared_ptr<Context> ctx,
+    ArrowSchema* schema,
+    ArrowSchema* spatial_schema,
+    ArrowArray* spatial_array,
+    const std::string& soma_type,
+    std::string_view type_metadata,
+    PlatformConfig platform_config) {
+    std::vector<Dimension> dims;
+    if (type_metadata.compare("WKB") != 0) {
+        throw TileDBSOMAError(std::format(
+            "[SOMAGeometryColumn] "
+            "Unkwown type metadata for `{}`: "
+            "Expected 'WKB', got {}",
+            SOMA_GEOMETRY_COLUMN_NAME,
+            type_metadata));
+    }
+
+    for (int64_t j = 0; j < spatial_schema->n_children; ++j) {
+        dims.push_back(ArrowAdapter::tiledb_dimension_from_arrow_schema(
+            ctx,
+            spatial_schema->children[j],
+            spatial_array->children[j],
+            soma_type,
+            type_metadata,
+            SOMA_GEOMETRY_DIMENSION_PREFIX,
+            "__min",
+            platform_config));
+    }
+
+    for (int64_t j = 0; j < spatial_schema->n_children; ++j) {
+        dims.push_back(ArrowAdapter::tiledb_dimension_from_arrow_schema(
+            ctx,
+            spatial_schema->children[j],
+            spatial_array->children[j],
+            soma_type,
+            type_metadata,
+            SOMA_GEOMETRY_DIMENSION_PREFIX,
+            "__max",
+            platform_config));
+    }
+
+    auto attribute = ArrowAdapter::tiledb_attribute_from_arrow_schema(
+        ctx, schema, type_metadata, platform_config);
+
+    return std::make_shared<SOMAGeometryColumn>(
+        SOMAGeometryColumn(dims, attribute.first));
+}
+
+void SOMAGeometryColumn::_set_dim_points(
+    const std::unique_ptr<ManagedQuery>& query,
+    const SOMAContext& ctx,
+    const std::any& points) const {
+    std::vector<std::pair<double_t, double_t>>
+        transformed_points = _transform_points(
+            std::any_cast<std::span<const std::vector<double_t>>>(points));
+
+    auto domain_limits = _limits(ctx, *query->schema());
```

Comment:
Core has current domain and domain. SOMA has domain and maxdomain.

Saying "domain" is ambiguous.

I've been consistent to say "core_domain", "core_current_domain"/"current_domain", "soma_domain", "soma_maxdomain". Please do the same here and make the variable name indicate whether this `domain` variable is intended to be core domain or soma domain (which are not the same).